react/components/ProductImages/index.js

```diff
@@ -72,27 +79,43 @@ const ProductImages = ({
     return showVideosFirst ? [...videos, ...images] : [...images, ...videos]
   }, [showVideosFirst, videos, images])
 
+  const { zoomType: legacyZoomType } = zoomProps || {}
+  const isZoomDisabled = legacyZoomType === 'no-zoom' || zoomMode === 'disabled'
+
   return (
-    <div
-      className={`${handles.productImagesContainer} ${handles.content} w-100`}
-    >
-      <Carousel
-        slides={slides}
-        position={position}
-        zoomMode={zoomMode}
-        maxHeight={maxHeight}
-        zoomFactor={zoomFactor}
-        aspectRatio={aspectRatio}
-        ModalZoomElement={ModalZoomElement}
-        thumbnailMaxHeight={thumbnailMaxHeight}
-        showPaginationDots={showPaginationDots}
-        thumbnailAspectRatio={thumbnailAspectRatio}
-        showNavigationArrows={showNavigationArrows}
-        thumbnailsOrientation={thumbnailsOrientation}
-        displayThumbnailsArrows={displayThumbnailsArrows}
-        // Deprecated
-        zoomProps={zoomProps}
-      />
+    <div className={`${productImagesContainerClass} ${handles.content} w-100`}>
+      {displayMode === 'inline' ? (
+        images.map(({ url, alt }, index) => (
+          <ProductImage
+            key={index}
+            src={url}
+            alt={alt}
+            maxHeight={maxHeight}
+            zoomFactor={zoomFactor}
+            aspectRatio={aspectRatio}
+            ModalZoomElement={ModalZoomElement}
+            zoomMode={isZoomDisabled ? 'disabled' : zoomMode}
+          />
+        ))
+      ) : (
+        <Carousel
+          slides={slides}
+          position={position}
+          zoomMode={zoomMode}
+          maxHeight={maxHeight}
+          zoomFactor={zoomFactor}
+          aspectRatio={aspectRatio}
+          ModalZoomElement={ModalZoomElement}
+          thumbnailMaxHeight={thumbnailMaxHeight}
+          showPaginationDots={showPaginationDots}
+          thumbnailAspectRatio={thumbnailAspectRatio}
+          showNavigationArrows={showNavigationArrows}
+          thumbnailsOrientation={thumbnailsOrientation}
+          displayThumbnailsArrows={displayThumbnailsArrows}
+          // Deprecated
+          zoomProps={zoomProps}
+        />
+      )}
```

Comment:
Nit: can you replace this ternary with an `if(...) { return ... } return ...`?  It's a tad more readable

Reply:
@kaisermann it is definitely possible, but, since both `<ProductImages />` and `<Carousel />` share a parent HTML Element `<div className={`${productImagesContainerClass} ${handles.content} w-100`}>` it would became a bit more complex in order to avoid redundancy, we would have to create a wrapper component like `<ProductImagesContainer />` and pass props around, etc... Are you ok with that?

Reply:
Can't we just store that className in a variable 😁 ?

Reply:
Sure!